store/src/java/com/zimbra/cs/html/owasp/OwaspHtmlSanitizer.java

```diff
@@ -79,6 +84,52 @@ private String checkUnbalancedTags(String html) {
         }
         return html;
     }
+    
+    /** 
+     * A method to check & remove any any child parent comparator inside <style> tag 
+     * 
+     * @return the Formatted Html after removing child comparators(>)
+     */
+    private String removeChildComaparator(String html) {
+        if (StringUtil.isNullOrEmpty(html)) {
+            return html;
+        }
+        ArrayList<String> cssTagsList = new ArrayList<String>();
+        // in cssTagsList array we can add multiple tags 
+        cssTagsList.add("tbody");
+        cssTagsList.add("tr");
+        cssTagsList.add("div");
+        String formattedHtml = "";
+        String startStyle = "<style";
+        String endStyle = "</style>";
+        String[] splitFromStyleStart = START_STYLE.split(html);
+        for (String checkEndStyle : splitFromStyleStart) {
+             if (checkEndStyle.toLowerCase().contains(endStyle.toLowerCase())) {
+                 String[] betweenCodeStyleTagArr = END_STYLE.split(checkEndStyle);
+                 String betweenCodeStyleTag = betweenCodeStyleTagArr[0];
+                 for (String cssTag : cssTagsList) {
+                      String formattedHtmlCSSTag = "";
+                      String childCssTag = ">" +cssTag+ ">";
+                      if (betweenCodeStyleTag.toLowerCase().contains(childCssTag.toLowerCase())) {
+                          String[] splitChildCssTag = Pattern.compile(childCssTag, Pattern.CASE_INSENSITIVE).split(betweenCodeStyleTag);
+                          for(int i = 0; i < splitChildCssTag.length; i++) {
+                              if (i == 0) {
+                                  formattedHtmlCSSTag = formattedHtmlCSSTag + splitChildCssTag[i];
+                              } else {
+                                  formattedHtmlCSSTag = formattedHtmlCSSTag + cssTag + ">" + splitChildCssTag[i];
+                              }
+                          }
+                          betweenCodeStyleTag = formattedHtmlCSSTag;
+                          formattedHtmlCSSTag = "";
+                      }
+                 }
+                 formattedHtml = formattedHtml + startStyle + betweenCodeStyleTag + endStyle + betweenCodeStyleTagArr[1];
```

Comment:
@rupeshtelus with this change child comparator seems to be getting removed, but it should add space in place of it.

Reply:
added space on betweenCodeStyleTag variable before css tags